### PR TITLE
Spare leases

### PR DESCRIPTION
**Pull request by sargon**
Instead of spare blocks handle spare leases. This should prevent 
instant block inquires after the first client.

Please review, the statistics code should be merged first.

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -297,7 +306,7 @@ int main(int argc, char** argv) {
     printf("-o CODE:LEN:P1. .. .Pn DHCP Option with code,len and #len chars in decimal\n");
     printf("-b BLKSIZEPOW          Power over two of block size\n");
     printf("-B TIMEOUT             Block timeout\n");
-    printf("-s SPAREBLKS           Amount of spare blocks\n");
+    printf("-s SPARELEASES         Amount of spare leases (max: 256)\n");
```
Comment (BenBE) [inline main.c]: Maximum of uint8_t is 255.

File: main.c [hunk 2]
```diff
@@ -261,7 +270,7 @@ int main(int argc, char** argv) {
       break;
 
     case 's':
-      config.spare_blocks_needed = (uint8_t)atoi(optarg);
+      config.spare_leases_needed = (uint8_t)atoi(optarg);
```
Comment (BenBE) [inline main.c]: Missing range check for user-supplied argument. Missing check for non-zero argument.

File: main.c (hunk 1 above)
Comment (sargon) [reply main.c]: Fixed

File: main.c (hunk 2 above)
Comment (sargon) [reply main.c]: Zero is a possible valid value.

**Pull request merged**